### DISCUSSION TITLE
Starting with  development/debug flag question

### DISCUSSION THREAD

**Discussion by dominikfhnw** [NightscoutFoundation/xDrip#3701] [General]
Hi all,
I'm trying to get started with xdrip or more generally with accessing my G7 sensor over BLE.
I have a thousand questions... but one very concrete one at first. There is the "d" flag in Ob1G5CollectionService. Can that be somehow turned on in the app, or is that compile-time only? (https://github.com/NightscoutFoundation/xDrip/blob/master/wear/src/main/java/com/eveningoutpost/dexdrip/services/Ob1G5CollectionService.java#L215)

I've also been sniffing BLE traffic with an nrf52840 dongle... does anyone know what the Dexcom vendor specific data in the advertisment packages mean? It seems to be stable over at least some days. 

It seems there is no master documentation on the protocol, so I've been mostly reading Ob1G5StateMachine and Ob1G5CollectionService. And I guess I also have to read some wixel xdrip source code, would be nice to get that code running on nrf hardware, looks like the original wixel hardware is not available anymore

What would be the best way to ask questions?

**Comment by Navid200** [comment]
This explains how you can activate extra logs for only two Java methods.  https://navid200.github.io/xDrip/docs/Dexcom/OB1Debug.html

You can do the same for any other Java method to make the logs verbose.

This explains how you can use xDrip as a collector from a G7.
https://navid200.github.io/xDrip/docs/Dexcom/G7.html

You can ask any questions about how to use xDrip here.

**Reply by dominikfhnw** [reply]
Thanks for the reply.
Yes I know about the extra logs, my question was with all the log statements starting with an "if (d)", like here: 
https://github.com/NightscoutFoundation/xDrip/blob/23c06fcecff54c1366576bad8c724c7bf6146450/wear/src/main/java/com/eveningoutpost/dexdrip/g5model/Ob1G5StateMachine.java#L151-L152

Is that "d" flag settable from the app itself, or is it compile-time only?                                                             

edit2: tried again to fix the link to the code... fingers crossed

**Reply by Navid200** [reply]
Sorry I misunderstood.

My understanding of a final variable is that it cannot be modified after once initialized.  Therefore, it can only be changed at compile time.
And my understanding of Java is quite limited!
